db/migrate/20221129194704_create_join_table_user_post.rb

```diff
@@ -0,0 +1,16 @@
+class CreateJoinTableUserPost < ActiveRecord::Migration[7.0]
+  def change
+    create_join_table :users, :posts, table_name: :likes do |t|
+      # t.index [:user_id, :post_id]
+      # t.index [:post_id, :user_id]
+      t.timestamps
+    end
+    create_join_table :users, :posts, table_name: :comments do |t|
+      # t.index [:user_id, :post_id]
+      # t.index [:post_id, :user_id]
+      t.text :text
+
+      t.timestamps
+    end
+  end
+end
```

Comment:
- [ ] You don't need join table instead make sure you have  2 different tables one for comments and one for likes as per the requirements.